frontend/model/notifications/nativeNotification.js

```diff
@@ -4,50 +4,112 @@ import sbp from '@sbp/sbp'
 // NOTE: since these functions don't modify contract state, it should
 //       be safe to modify them without worrying about version conflicts.
 
-export async function requestNotificationPermission (force: boolean = false): Promise<null | string> {
-  if (typeof Notification === 'undefined') {
-    return null
-  }
+const handler = (statuses: string[]) => {
+  // For some reason, Safari seems to always return `'prompt'` with
+  // `Notification.permission` being correct.
+  const granted = statuses.every(status => status === 'granted') || (statuses.every(status => status === 'prompt') && Notification.permission === 'granted')
+  sbp('state/vuex/commit', 'setNotificationEnabled', granted)
+}
 
-  if (force || Notification.permission === 'default') {
-    try {
-      sbp('state/vuex/commit', 'setNotificationEnabled', await Notification.requestPermission() === 'granted')
-    } catch (e) {
-      console.error('requestNotificationPermission:', e.message)
-      return null
+const fallbackChangeListener = () => {
+  if (!Notification.permission) return
+  let oldValue = Notification.permission
+  handler([oldValue])
+  setInterval(() => {
+    const newValue = Notification.permission
+    if (oldValue !== newValue) {
+      oldValue = newValue
+      handler([oldValue])
     }
-  }
+  }, 100)
+}
+
+export const setupNativeNotificationsListeners = () => {
+  if (typeof navigator !== 'object' || typeof Notification !== 'function' || typeof PushManager !== 'function' || typeof ServiceWorker !== 'function' || typeof navigator.serviceWorker !== 'object') return
```

Comment:
Claude:

```js
// Current check:
if (typeof navigator !== 'object' || 
    typeof Notification !== 'function' || 
    typeof PushManager !== 'function' || 
    typeof ServiceWorker !== 'function' || 
    typeof navigator.serviceWorker !== 'object') return

// ServiceWorker check might not be necessary since we have a fallback
```

But mostly I'm copying this because this line is way too long and Claude formatted it in a better way.